freeform_json_to_yaml.py

```diff
@@ -0,0 +1,440 @@
+        'weather': 'clear',
+        'lighting': 'bright',
+        'visibility': 'excellent',
+        'noise_peak': 'noisy',
+        'temperature': 92,
+        'fauna': 'normal'
+    },
+    "jungle": {
+        'type': 'jungle',
+        'terrain': 'jungle',
+        'temperature': 88,
+        'humidity': 90,
+        'visibility': 'low',
+        'flora': 'lush'
+    },
+    "submarine": {
+        'type': 'submarine',
+        'terrain': 'indoors',
+        'visibility': 'low',
+        'fauna': 'none',
+        'flora': 'none'
+    },
+    "urban": {
+        'type': 'urban',
+        'terrain': 'urban',
+        'lighting': 'normal',
+        'visibility': 'moderate',
+        'noise_ambient': 'noisy',
+        'flora': 'none',
+        'fauna': 'none'
+    }
+}
+
+SEVERITY_MAP = {
+    "None": "minor",
+    "Smallest": "minor",
+    "Small": "moderate",
+    "Medium": "substantial",
+    "Large": "major",
+    "Largest": "extreme"
+}
+
+POSE_MAP = {
+    "standing": "standing",
+    "sittingGround": "sitting on the ground",
+    "sittingChair": "sitting on a chair",
+    "kneeling": "kneeling",
+    "supine": "in a supine position",
+    "recovery": "in a recovery position on the ground",
+    "fetal": "in a fetal position on the ground",
+    "prone": "in a prone position on the ground",
+    "shooting": "actively shooting"
+}
+
+class JsonConverter:
+    json_data = {}
+    output_dest = None
+
+    def __init__(self, input_path, output_path):
+        f = None
+        try:
+            f = open(input_path, 'r', encoding='utf-8')
+            self.json_data = json.load(f)
+            f.close()
+        except Exception as e:
+            if f is not None:
+                f.close()
+            LOGGER.log(LogLevel.FATAL, "Error while loading in json file -- " + str(e))
+        
+        try:
+            if output_path[-5:] != '.yaml':
+                LOGGER.log(LogLevel.FATAL, "Error while setting up output location -- output must be a .yaml file")
+            self.output_dest = open(output_path, 'w', encoding='utf-8')
+        except Exception as e:
+            LOGGER.log(LogLevel.FATAL, "Error while setting up output location -- " + str(e))
+
+
+    def __del__(self):
+        if self.output_dest is not None:
+            self.output_dest.close()
+
+
+    def convert(self):
+        yaml_data = {}
+        yaml_data['id'] = self.json_data['scenarioData']['name']
+        yaml_data['name'] = self.json_data['scenarioData']['description']
+        sim_env = ENV_MAP[self.json_data['scene'].split('-')[1].replace('sub', 'submarine')]
+        yaml_state = {}
+        unstructured = 'TODO'
+        narrative_sections = self.json_data.get('narrative', {}).get('narrativeSections', [])
+        for x in narrative_sections:
+            if 'additionalInfo' in x:
+                # get the first additional info available for unstructured text
+                unstructured = x['additionalInfo']
+                break
```

Comment:
This is really cool, but the only problem is that ADMs (read: Parallax) will want/need supporting structured data for whatever is in the `narrativeSections.additionalInfo`, which I don't think we'll be able to generate.

Given that these are for freeform / open world scenarios, do we need to get this from the JSON or can we just use your generic boilerplate text?

Reply:
Brian wants it from the JSON. If we need to manually fine-tune it later, I don't see why that would be a problem. I think the point of this is just to make the process easier. We all know that we can't go back and forth between files with all the details we want perfectly

Reply:
Sounds good...as long as we add the structured data ourselves, then we're all good.